eessi/reframe/eessi_utils/hooks.py

```diff
@@ -1,37 +1,32 @@
+"""
+Hooks for setting job resources in ReFrame tests
+"""
+
 import reframe as rfm
-import eessi_utils.utils as utils
+from eessi_utils import utils
 
-processor_info_missing = '''This test requires the number of CPUs to be known for the partition it runs on.
+PROCESSOR_INFO_MISSING = '''This test requires the number of CPUs to be known for the partition it runs on.
 Check that processor information is either autodetected
-(see https://reframe-hpc.readthedocs.io/en/stable/configure.html#proc-autodetection),
-or manually set in the ReFrame configuration file
-(see https://reframe-hpc.readthedocs.io/en/stable/config_reference.html?highlight=processor%20info#processor-info).
+    (see https://reframe-hpc.readthedocs.io/en/stable/configure.html#proc-autodetection),
+    or manually set in the ReFrame configuration file
+    (see https://reframe-hpc.readthedocs.io/en/stable/config_reference.html#processor-info).
 '''
 
 
-def skip_cpu_test_on_gpu_nodes(test: rfm.RegressionTest):
-    '''Skip test if GPUs are present, but no CUDA is required'''
-    skip = (utils.is_gpu_present(test) and not utils.is_cuda_required(test))
-    if skip:
-        test.skip_if(True, f"GPU is present on this partition ({test.current_partition.name}), skipping CPU-based test")
-
-
-def skip_gpu_test_on_cpu_nodes(test: rfm.RegressionTest):
-    '''Skip test if CUDA is required, but no GPU is present'''
-    skip = (utils.is_cuda_required(test) and not utils.is_gpu_present(test))
-    if skip:
-        test.skip_if(
-            True,
-            f"Test requires CUDA, but no GPU is present in this partition ({test.current_partition.name}). "
-            "Skipping test..."
-        )
-
-
 def assign_one_task_per_feature(test: rfm.RegressionTest, feature) -> rfm.RegressionTest:
```

Comment:
I'm still not sure if 'feature' is the most descriptive word here. I know you're referring to the features from the ReFrame config, but it feels... non-descriptive. 

Something like `..._per_compute_unit(test: rfm.RegressionTest, compute_unit)` also sounds a bit strange, but is maybe already more clear? In theory, we could then even expand that function to take e.g. a socket as a unit (makes sense for e.g. TensorFlow).

Reply:
good point, it doesn't have to be a feature that is added as an argument, in theory any string is valid.